From e29a7f31aec2e9c9d5881fd115b1f62724cacd95 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E5=BF=98=E6=80=80?= <otgwt@outlook.com>
Date: Wed, 13 Nov 2024 18:38:37 +0800
Subject: [PATCH 231/360] net: ethernet: phytmac: Convert to platform remove
 callback returning void
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

0edb555: platform: Make platform_driver::remove() return void
cause build error, so convert .remove from int to void

drivers/net/ethernet/phytium/phytmac_platform.c:243:19: error: initialization of ‘void (*)(struct platform_device *)’ from incompatible pointer type ‘int (*)(struct platform_device *)’ [-Werror=incompatible-pointer-types]
  243 |         .remove = phytmac_plat_remove,
      |                   ^~~~~~~~~~~~~~~~~~~
drivers/net/ethernet/phytium/phytmac_platform.c:243:19: note: (near initialization for ‘phytmac_driver.<anonymous>.remove’)
---
 drivers/net/ethernet/phytium/phytmac_platform.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/drivers/net/ethernet/phytium/phytmac_platform.c b/drivers/net/ethernet/phytium/phytmac_platform.c
index 343e1a5ce564..985ae83b3977 100644
--- a/drivers/net/ethernet/phytium/phytmac_platform.c
+++ b/drivers/net/ethernet/phytium/phytmac_platform.c
@@ -204,14 +204,12 @@ static int phytmac_plat_probe(struct platform_device *pdev)
 	return ret;
 }
 
-static int phytmac_plat_remove(struct platform_device *pdev)
+static void phytmac_plat_remove(struct platform_device *pdev)
 {
 	struct phytmac *pdata = platform_get_drvdata(pdev);
 
 	phytmac_drv_remove(pdata);
 	phytmac_free_pdata(pdata);
-
-	return 0;
 }
 
 static int __maybe_unused phytmac_plat_suspend(struct device *dev)
-- 
2.20.1

